src/peer.ts

```diff
@@ -1,15 +1,194 @@
-import { IpRegistry } from "./ipRegistry";
+import { logger } from "@blockr/blockr-logger";
+import { Guid } from "guid-typescript";
+
+import { MessageType } from "./enums";
+import { IMessageListener } from "./interfaces/iMessageListener";
+import { IPeer } from "./interfaces/iPeer";
+import { Message } from "./models/message";
 import { Receiver } from "./receiver";
+import { RoutingTable } from "./routingTable";
 import { Sender } from "./sender";
+import { DateManipulator } from "./util/dateManipulator";
+
+const MESSAGE_EXPIRATION_TIMER: number = 1;
+const MESSAGE_HISTORY_CLEANUP_TIMER: number = 60000; // One minute
+
+/**
+ * Handles the peer network.
+ */
+export class Peer implements IMessageListener, IPeer {
+    private readonly routingTable: RoutingTable;
+    private readonly receiveHandlers: Map<string, (message: Message, senderIp: string) => void>;
+    private readonly sender: Sender;
+    private readonly receiver: Receiver;
+    private GUID: string;
+
+    constructor(port: string, initialPeers?: string[]) {
+        this.routingTable = new RoutingTable();
+        this.receiveHandlers = new Map();
+        this.createReceiverHandlers();
+
+        this.sender = new Sender(port);
+        this.receiver = new Receiver(this, port);
+
+        // Create timer that removes peers that did not reply
+        setInterval(() => {
+            const minDate = DateManipulator.minusMinutes(new Date(), MESSAGE_EXPIRATION_TIMER);
+            this.sender.getSentMessagesSendersSince(minDate).forEach((value: string) => {
+                this.routingTable.removePeer(value);
+                logger.info(`Peer removed from routing table: ${value}`);
+            });
+        }, MESSAGE_HISTORY_CLEANUP_TIMER);
+
+        // If initialPeers is undefined, this instance is the first peer
+        if (initialPeers === undefined) {
```

Comment:
Replace `if (initialPeers === undefined)` with `if (initialPeers)` because this also checks if the variable is undifined